### ISSUE TITLE
Needed more VLines updated int_to_roman method in core.util

### ISSUE THREAD

**Issue by neuronist** [ioam/holoviews#2353]
```
def int_to_roman(input):
    if type(input) != type(1):
        raise TypeError("expected integer, got %s" % type(input))
    if not 0 < input < 4000000:
        raise ValueError("Argument must be between 1 and 3999999")
    ints = (1000000, 500000, 100000, 50000, 10000, 5000, 1000, 900,  500, 400, 100,  90, 50,  40, 10,  9,   5,  4,   1)
    nums = ('M_', 'D_', 'C_', '', 'x', 'v', 'M',  'CM', 'D', 'CD','C', 'XC','L','XL','X','IX','V','IV','I')
    result = ""
    for i in range(len(ints)):
        count = int(input / ints[i])
        result += nums[i] * count
        input -= ints[i] * count
    return result
```
I'm overlaying multiple VLines and needed more than 3999 VLines in my plot so I increased the threshold to 3999999, removing the need for roman numerals might be beneficial in the long run. This is just a simple solution that worked for my specific case. Lowercase letters roman numerals for this case represent the higher numbers (based on this [table](https://www.rapidtables.com/math/symbols/roman_numerals.html)) might not be that accurate but worked for what I needed.

**Comment by philippjfr** [comment]
Seems like that's going to highly inefficient, the better solution is probably to support vectorized VLine and HLine elements, which I'd be in favor of.

**Comment by neuronist** [comment]
I completely agree with you. I'm working with data that is only plotable with datashader, in this case even when I increased the limit of roman numerals, the plot was too slow (zooming and panning) and I ended up downsampling the number of VLines I wanted to plot.

**Comment by jbednar** [comment]
And you can't use the Spikes element in hv, which should translate to a datashader line plot?

**Comment by philippjfr** [comment]
Unfortunately I don't think bokeh actually has vectorized  ``Span`` glyph, which means that even if we were to vectorize it at the HoloViews level the actual performance when zooming on the plot would be unchanged.

>And you can't use the Spikes element in hv, which should translate to a datashader line plot?

I'm not actually sure the datashader operation understands Spikes yet, but that should be an easy addition.

**Comment by neuronist** [comment]
Might be possible, didn't try and I'm not at the laptop right now. Thanks for the suggestion! 

But are you able to color Spikes?
I wanted to plot VLines that represent specific events on a time series, coloring them depending on the event is interesting for an easier visualization of the sequence of events and behavior of the curve.

**Comment by jbednar** [comment]
@jlstevens, can you comment here?  I think you were using datashaded spikes at some point, but (a) I don't remember if you colorized it, and (b) I don't know if that was on a branch or whether it got checked in.

**Comment by jlstevens** [comment]
I colorized the spikes but not individually. Depending on how many categories there are that need coloring, you could maybe filter the spikes by type then overlay the spikes (each set of spikes having the color you want). You definitely do want to use the ``Spikes`` element here instead of overlaying thousands of ``VLine`` elements! :-)

**Comment by LocusBio** [comment]
I just ran into this issue when annotating a plot with a large number of text items. I applied the patch to the code listed above, and while the annotations no longer error out, it is clear that the ones past 4000 aren't rendering. Seems like an odd bound to place on things. How might we go about increasing the relatively arbitrary number of elements per plot to allow for more?
